TheMET/TheMET/ArtFilter/ArtFilter.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  ArtFilter.swift
+//  TheMET
+//
+//  Created by Анна Ситникова on 10/08/2023.
+//
+
+import Foundation
+import UIKit
+
+class ArtFilter {
+    func filter(arts: [Art], searchText: String?) -> [Art] {
+        var filteredArts:[Art] = []
+        guard let searchText = searchText else {
+            return arts
+        }
+        for art in arts {
+            if self.filter(artProperty: art.artistDisplayName, searchText: searchText) ||
+                self.filter(artProperty: art.department, searchText: searchText) ||
+                self.filter(artProperty: art.medium, searchText: searchText) ||
+                self.filter(artProperty: art.objectName, searchText: searchText) ||
+                self.filter(artProperty: art.title, searchText: searchText) {
+                guard !filteredArts.contains(where: { filteredArt in
+                    filteredArt.objectID == art.objectID
+                }) else {
+                    return filteredArts
```

Comment:
You're executing `return` in the middle of the `for`-cycle. It means some elements of the original array are nit checked yet and won't be included to the result

Reply:
+